velox/exec/Task.h

```diff
@@ -640,6 +640,31 @@ class Task : public std::enable_shared_from_this<Task> {
     return terminateRequested_;
   }
 
+  /// Requests barrier on task execution to drain out all the buffered data
+  /// after finishes processing all the queued source splits. The returned
+  /// future will be realized when the drain operations finish. This is only
+  /// supported on task under single threaded execution mode and all its plan
+  /// nodes support barrier processing.
+  ContinueFuture requestBarrier();
+
+  /// Returns true if this task is under barrier processing.
+  bool underBarrier() const {
+    return barrierRequested_;
+  }
+
+  /// Sets to suppress the input for the specified operator. 'planNodeId' is the
+  /// associated plan node id. The function is called when a draining operator
+  /// needs no more input to finish the drain operation. All the upstream
+  /// operators can simply drop their output and output processing to finish the
+  /// drain quickly.
+  void suppressInput(const core::PlanNodeId& planNodeId, size_t operatorId);
+
+  /// Invoked by a driver thread to signal it has finished the barrier
+  /// processing when all its operators have drained their output and the sink
+  /// operator has propogated the drain signal to all its consumer pipelines
+  /// through the connected queues.
+  void finishDriverBarrier();
```

Comment:
Perhaps, clarify that each Driver is expected to call this once and explain what happens when last Driver calls this API.